huaweicloud/services/acceptance/dli/resource_huaweicloud_dli_flinkjar_job_test.go

```diff
@@ -0,0 +1,237 @@
+  log_enabled     = true
+}
+`, ak, sk, jarObsPath, bucketName, name, name, region, name)
+}
+
+func TestAccResourceDliFlinkJarJob_all(t *testing.T) {
+	var obj flinkjob.CreateJarJobOpts
+	resourceName := "huaweicloud_dli_flinkjar_job.test"
+	name := acceptance.RandomAccResourceName()
+	bucketName := acceptance.RandomAccResourceNameWithDash()
+
+	rc := acceptance.InitResourceCheck(
+		resourceName,
+		&obj,
+		getDliFlinkJarJobResourceFunc,
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			acceptance.TestAccPreCheck(t)
+			acceptance.TestAccPreCheckDliJarPath(t)
+		},
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      rc.CheckResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccFlinkJarJobResource_all(name, bucketName, acceptance.HW_REGION_NAME, acceptance.HW_ACCESS_KEY,
+					acceptance.HW_SECRET_KEY, acceptance.HW_DLI_FLINK_JAR_OBS_PATH),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(resourceName, "name", name),
+					resource.TestCheckResourceAttr(resourceName, "status", "job_running"),
+					resource.TestCheckResourceAttr(resourceName, "queue_name", name),
+					resource.TestCheckResourceAttrPair(resourceName, "obs_bucket", "huaweicloud_obs_bucket.test", "bucket"),
+					resource.TestCheckResourceAttr(resourceName, "log_enabled", "true"),
+					resource.TestCheckResourceAttr(resourceName, "description", "description of dli job"),
+					resource.TestCheckResourceAttr(resourceName, "main_class", "com.huaweicloud.dli.obs.Main"),
+					resource.TestCheckResourceAttr(resourceName, "feature", "basic"),
+					resource.TestCheckResourceAttr(resourceName, "flink_version", "1.10"),
+					resource.TestCheckResourceAttr(resourceName, "cu_num", "6"),
+					resource.TestCheckResourceAttr(resourceName, "manager_cu_num", "2"),
+					resource.TestCheckResourceAttr(resourceName, "parallel_num", "2"),
+					resource.TestCheckResourceAttr(resourceName, "tm_slot_num", "1"),
+					resource.TestCheckResourceAttr(resourceName, "smn_topic", name),
+					resource.TestCheckResourceAttr(resourceName, "restart_when_exception", "true"),
+					resource.TestCheckResourceAttr(resourceName, "resume_max_num", "2"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccFlinkJarJobResource_all(name, bucketName, region, ak, sk, jarObsPath string) string {
+	return fmt.Sprintf(`
+variable "ak" {
+  type        = string
+  description = "value"
+  default     = "%s"
+}
+
+variable "sk" {
+  type        = string
+  description = "value"
+  default     = "%s"
+}
+
+variable "jarObsPath" {
+  type        = string
+  description = "value"
+  default     = "%s"
+}
+
+resource "huaweicloud_obs_bucket" "test" {
+  bucket        = "%s"
+  acl           = "private"
+  force_destroy = true
+}
+
+resource "huaweicloud_dli_package" "test" {
```

Comment:
can we use the result of `testAccDliPackage_basic`? then we can delete **TestAccPreCheckDliJarPath** function

Reply:
no, this jar is a real flink job jar which will run after submit job. 
testAccDliPackage_basic just test whether the file is uploaded